site/src/pages/release-notes/1.19.0.mdx

```diff
@@ -0,0 +1,189 @@
+        from("com.linecorp.armeria:armeria-version-catalog:1.19.0")
+      }
+    }
+  }
+  ```
+
+## 📈 Improvements
+
+- `CompositeException` now shows full stack traces for all nested exceptions 
+  when `-Dcom.linecorp.armeria.verboseExceptions=true` is set. #4347 #4356
+
+## 🛠️ Bug fixes
+
+- You can now use Armeria Spring actuator with WebFlux correctly. #4396 #4408
+- You no longer see `ClassCastException` when an invalid path request is received. #4405
+- Annotated services now correctly return `204 No Content` status when `void`, `Publisher<Void>` or
+  `CompletionStage<Void>` and `@ProducesJson` are used at the same time. #4390 #4398
+  ```java
+  // 200 OK status and "null" string body is returned
+  @Get("/null-content")
+  @ProducesJson
+  public Publisher<Void> nullBody() {
+    return Mono.empty();
+  }
+  
+  // 204 No Content status and no body is returned
+  @Get("/no-content")
+  @StatusCode(204)
+  @ProducesJson
+  public Publisher<Void> noBody() {
+    return Mono.empty();
+  }
+  ```
+- Armeria server no longer sends chunked encoding header with `304 Not Modified` status. #4386
+- HTTP/2 graceful shutdown timeout is correctly set using <type://ClientFactory#idleTimeoutMillis()>. #4381
+- Unframed <typeplural://GrpcService> now support `application/x-protobuf` and `application/x-google-protobuf`
+  media types. #4355 #4364
+- You can now use a custom `JsonFormat.Printer` for <type://ProtobufResponseConverterFunction>  #4301 #4336
+
+## 📃 Documentation
+
+- You can now learn how to write a gRPC service and client with Armeria by walking through
+   [the gRPC tutorial](/tutorials/grpc/blog). #3987
+  - Special thanks to @freevie who volunteered for this.
+
+## 🏚️ Deprecations
+
+- <type://DnsResolverGroupBuilder#refreshBackoff(Backoff)> has been deprecated in favor of
+  <type://DnsResolverGroupBuilder#autoRefreshBackoff(Backoff)>. #4400
+
+## ☢️ Breaking changes
+
+- The method signature of <type://ResponseConverterFunctionProvider#createResponseConverterFunction(Type)> has
+  been changed.
+  - <type://ResponseConverterFunctionProvider> no longer takes a <typeplural://ResponseConverterFunction> as 
+    the second parameter. Use <type://DelegatingResponseConverterFunctionProvider> to create a new 
+    <type://ResponseConverterFunction> based on the delegating <typeplural://ResponseConverterFunction>.
+
+## ⛓ Dependencies
+
+- Brave 5.13.10 → 5.13.11
+- Brotli4J 1.7.1 → 1.8.0
+- Dropwizard Metrics 4.2.10 → 4.2.12
+- gRPC-Java 1.48.0 → 1.49.0
+- Jackson 2.13.3 → 2.13.4
+- Logback 1.2.11 → 1.3.0
+- Micrometer 1.9.2 → 1.9.3
+- Reactor 3.4.21 → 3.4.22
+- Sangria 3.0.1 → 3.2.0
+- Scala 3.1.3 → 3.2.0
+- Spring Boot 2.7.2 → 2.7.3
+
+## 🙇 Thank you
+
+<ThankYou usernames={[
+  'DasAmpharos',
+  'anuraaga',
+  'cnabro',
+  'devdynam0507',
+  'dinujoh',
+  'freevie',
+  'ghkim3221',
+  'heowc',
+  'ikhoon',
+  'injae-kim',
+  'j-min5u',
+  'jrhee17',
+  'klurpicolo',
+  'ks-yim',
+  'minwoox',
+  'resquivel-squareup',
```

Comment:
Can you also add [tawAsh1](https://github.com/tawAsh1) for reporting the bug on `ClassCastException`?

Reply:
Thanks for noting it. I missed him.